pdr-api/handlers/parks/_identifier/name/PUT/index.js

```diff
@@ -0,0 +1,199 @@
+ */
+async function minorUpdate(user, body, currentTimeISO) {
+  // Calls the 'updateRecord' function to update the record.
+  const attributes = await updateRecord(user, body, currentTimeISO);
+
+  // Returns a success response with the updated attributes.
+  return sendResponse(200, attributes, 'Record updated');
+}
+
+/**
+ * Handles a legal name change.
+ *
+ * @param {string} user - The user making the update.
+ * @param {Object} body - The request body.
+ * @param {string} currentTimeISO - The current time in ISO format.
+ * @returns {Promise<Object>} - A Promise that resolves to the response object.
+ */
+async function majorChange(user, body, currentTimeISO) {
+  // Creates a changelog item for the legal name change.
+  await createChangeLogItem(body, currentTimeISO);
+
+  // Calls the 'updateRecord' function to update the record.
+  const attributes = await updateRecord(user, body, currentTimeISO);
+
+  // Returns a success response with the updated attributes.
+  return sendResponse(200, attributes, 'Record updated', 'Legal Name Change');
+}
+
+/**
+ * Creates a changelog item for a legal name change.
+ *
+ * @param {string} user - The user making the update.
+ * @param {Object} body - The request body.
+ * @param {string} currentTimeISO - The current time in ISO format.
+ * @returns {Promise<void>} - A Promise that resolves when the changelog item is created.
+ */
+async function createChangeLogItem(body, currentTimeISO) {
+  // Logs the creation of a changelog item.
+  logger.info('Creating changelog item');
+
+  // Retrieves the current record based on the 'orcs' identifier.
+  const currentRecord = await getOne(body.orcs, 'Details');
+
+  // Converts the current record to a format suitable for DynamoDB.
+  let changelogRecord = AWS.DynamoDB.Converter.marshall(currentRecord);
+
+  // Sets the 'sk' property of the changelog record to the current time in ISO format.
+  changelogRecord['sk'] = { 'S': currentTimeISO };
+
+  // This will copy the incomign legal, effective, and status as a 'new' item to ensure
+  // we record the switch from->to regardless whether 1 or more attributes are changing as
+  // part of this majorChange update.
+  changelogRecord['newLegalName'] = { 'S': body.legalName };
+  changelogRecord['newEffectiveDate'] = { 'S': body.effectiveDate };
+  changelogRecord['newStatus'] = { 'S': body.status };
+
+  // Calls the 'putItem' function to store the changelog record in DynamoDB.
+  await putItem(changelogRecord);
+}
+
+/**
+ * Updates a record in DynamoDB.
+ *
+ * @param {string} user - The user making the update.
+ * @param {Object} body - The request body.
+ * @param {string} currentTimeISO - The current time in ISO format.
+ * @returns {Promise<Object>} - A Promise that resolves to the updated attributes.
+ */
+async function updateRecord(user, body, currentTimeISO) {
+  // Defines the parameters for updating the record.
+  let updateParams = {
+    TableName: TABLE_NAME,
+    Key: {
+      pk: { S: body.orcs },
```

Comment:
We should be getting the `orcs` from the `:identifier` URL path variable - otherwise you can POST to one orcs with the post body being set to another.

Reply:
Fixed in d72ef9a